jekyll/_cci2/using-orbs.md

```diff
@@ -0,0 +1,158 @@
+
+```
+jobs:
+  my-job:
+    executor:
+      name: my-executor
+    steps:
+      -run: echo outside the executor
+```
+
+
+
+## Key Concepts
+
+Before using Orbs, you should first familiarize yourself with some basic core concepts of Orbs and how they are structured and operate. Gaining a basic understanding of these core concepts will enable you to leverage Orbs and use them easily in your own environments.
+
+### Development vs. Production Orbs
+{:.no_toc}
+
+Orbs can be published either as ```dev:foo``` or as a semantically versioned production orb. Development Orbs are mutable and expire, whereas Production Orbs are immutable and durable.
+
+### Certified vs. 3rd Party Orbs
+{:.no_toc}
+
+CircleCI will make available a number of individual Orbs that have been tested and certified to work with the platform. These orbs will be treated as part of the platform; all other orbs are considered 3rd party Orbs.
+
+<aside class="notice">
+All Orbs are open, meaning that anyone can use them and see their source. Also, only the ```circleci``` namespace is certified.
+</aside>
+
+## Design Methodology
+
+Before using orbs, you may find it helpful to understand the various design decisions and methodologies that were used when these Orbs were designed. Orbs were designed with the following considerations:
+
+* Orbs are transparent - If you can execute an Orb, you can view the source of that Orb.
+* Metadata is baked in - Everything can take a ```description``` key.
+* Orbs are ALWAYS semantic versioned (semver'd) - CircleCI allows development Orbs that have versions that start with `dev:`.
+* Production Orbs are immutable - Once an Orb has been published to a semantic version, the Orb cannot be changed. This prevents unexpected breakage or changing behaviors in core orchestration.
+* One registry (per install) - Each installation of CircleCI, including circleci.com, has only one registry where Orbs can be kept.
+* Organization Admins publish production Orbs. Organization members publish development Orbs - All namespaces are owned by an organization. Only the admin(s) of that organization can publish/promote a production Orb. All organization members can publish development Orbs.
+
+### Designing Orbs
+
+{:.no_toc}
+
+When designing your own Orbs, make sure your Orbs meet the following requirements:
+
+* Orbs should always use 'description'. - Be sure to explain usage, assumptions, and any tricks in the ```description``` key under jobs, commands, executors, and parameters.
+* Match commands to executors - If you are providing commands, try to provide one or more executors in which they will run.
+* Use concise naming for your Orb - Remember that use of your commands and jobs is always contextual to your Orb, so you can use general names like "run-tests" in most cases.
```

Comment:
It's also helpful to include the type in the name so it's more legible at the call site, but I'm not sure we want to set that as best practice. @ndintenfass do you have a preference?

e.g. `hannah/foo@dev:bar` could have a command `baz-cmd` and an executor `baz-exec` so, when i call them it looks like:

```
version: 2.1

orbs:
    foo: hannah/foo@dev:bar

jobs:
    baz:
       executor: foo/baz-exec
       steps:
         - foo/baz-cmd
```

Reply:
I'm actually not in favor of suggesting you "type" the name of an orb element. I'm totally fine with the same name for different elements because the semantic nature of the DSL allows it on purpose.

Reply:
Fair enough, that makes sense to me.

Reply:
leaving as-is for now